ledger/block/src/transaction/bytes.rs

```diff
@@ -17,7 +17,9 @@ use super::*;
 impl<N: Network> FromBytes for Transaction<N> {
     /// Reads the transaction from the buffer.
     #[inline]
-    fn read_le<R: Read>(mut reader: R) -> IoResult<Self> {
+    fn read_le<R: Read>(reader: R) -> IoResult<Self> {
+        // Restrict the reader to `MAX_TRANSACTION_SIZE` bytes.
+        let mut reader = reader.take(N::MAX_TRANSACTION_SIZE as u64);
```

Comment:
I echo @iamalwaysuncomfortable. I don't know if using `take` and truncating the bytestream is the best approach. However, I'm also not an expert on the implications.

Could @joske or @niklaslong chime in here? The intention is to limit transactions to 128kb in consensus, in order to ensure transactions exceeding this size don't land in a block. At the same time, we also want to minimize any risk of breakages to any other functionality in our codebase.

The priority is on safety over fast or simple results here.

Reply:
I agree with the above. One consideration is DoS vectors in snarkOS, however we already limit inbound message sizes in the codecs so we should be covered there. That means we're left with the problem of detecting overly large transactions that are _smaller_ than the network's max message size as early as possible and I think that would be safe to check after deserialisation. 

Is there any other path in transaction processing we don't cover that would be relevant here?

Reply:
> Just for completeness and in order to avoid decoding overly large transactions in snarkOS, we could adjust the codecs to:
> 
>     1. register the length of the raw bytes from the network
> 
>     2. read the first two bytes containing the message ID
> 
>     3. if it is a tx and longer than the max, we bail, if not we decode the rest

I'd be in favor of this approach as the `MAX_EVENT_SIZE` is 256 MiB, while the `MAX_TRANSACTION_SIZE` is 128 kB.

Reply:
We could restrict serialisation in the same way but I think this should be a property enforced in snarkVM? i.e. an honest validator can't produce a tx that is too large in the first place? Or is it impossible to check for before serialisation?

Reply:
Ah, I misunderstood the question. When it comes to _signing_ other proposals that might have large transactions in them, the assumption is that we've seen and stored the tx so restricting inbound size should already account for that. However, we might need to add a little nuance to the logic and error handling 🤔

Reply:
To add to the above, we check that transactions are well formed in the proposal so we could maybe check for length there (we also deserialise txs at that stage).